presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -605,7 +606,12 @@ public void run()
                         }
                     }
                     catch (Throwable t) {
-                        log.error(t, "Error processing %s", split.getInfo());
+                        if (t instanceof PrestoException) {
+                            log.error("Error processing %s: %s", split.getInfo(), t);
+                        }
+                        else {
```

Comment:
This is a catch-all for any exception that happens while processing a task. These could be unexpected errors, in which case we want to log a stack trace, or they could be `PrestoException`, in which case we assume the cause is known and we don't need to clutter the log with a stack trace.